pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -130,6 +131,32 @@ public MessageId send() throws PulsarClientException {
         return this;
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public TypedMessageBuilder<T> loadConf(Map<String, Object> config) {
+        config.forEach((key, value) -> {
+            if (key.equals("key")) {
```

Comment:
Can we define CONSTANTS for those keys? It is better to use CONSTANTS to avoid any typos when using this approach.